File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
      int current = 0;
      for(int i=0 ; i<nums.size();i++){
        if(nums[i] != 0){
            nums[current++] = nums[i];
        }
      }
        while(current < nums.size()){
            nums[current++] = 0;
        }
      }
};

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void swap(int [] num ,int i , int j){
        int temp = num[i] ;
        num[i] = num[j];
        num[j] = temp;
    }
    public void moveZeroes(int[] nums) {
        int left = 0;
        for(int right = 0 ; right < nums.length ;right++){
            if(nums[right]!= 0){
                swap(nums,left,right);
                left++;
            }
        }
    }
}